src/nvim/api/private/helpers.c

```diff
@@ -808,6 +811,17 @@ static void init_type_metadata(Dictionary *metadata)
 
   PUT(*metadata, "types", DICTIONARY_OBJ(types));
 }
+static void init_api_level_metadata(Dictionary *metadata)
+{
+  Dictionary version = ARRAY_DICT_INIT;
+  PUT(version, "current", INTEGER_OBJ(NVIM_API_CURRENT));
+  PUT(version, "compatibility", INTEGER_OBJ(NVIM_API_COMPATIBILITY));
+#ifdef NVIM_VERSION_PRERELEASE
+  PUT(version, "prerelease", BOOLEAN_OBJ(true));
```

Comment:
The point is that (the state I want to call) `[2, "pre"]` is neither the same as `[1, "stable"]` nor `[2, "stable"]`. If we don't include it, we need to make a decision one way or the other. You seem to prefer the latter. But then there is no simple way to determine "what is the latest stable version?" 

> in either case you need to verify that the stable API is the same as the last snapshot you used of the prerelease API.

I suppose most plugins/hosts, no matter the strategy towards master versions, could _in principle_ make a decision purely by inspecting the detailed method metadata, _without_ using the global version data at all at runtime. My point is simply that _both_ plugins that wants to only use stable _and_ those that wants to try use unstable api level (if available) should be allowed to do so based on the global fields without deep inspection. Saying that it "only be useful" for one or the other, is missing the big picture IMHO.

 If one only imagines one plugin then the granularity might not make sense. The point is rather that different plugins might want to put the decision boundary differently, even if it is likely no single plugin will ever alone distinguish all of 1-stable, 2-pre and 2-release.